src/java.base/share/classes/java/lang/String.java

```diff
@@ -974,9 +974,12 @@ private static byte[] getBytesNoRepl1(String s, Charset cs) {
     private static byte[] encodeASCII(byte coder, byte[] val) {
         if (coder == LATIN1) {
             byte[] dst = Arrays.copyOf(val, val.length);
-            for (int i = 0; i < dst.length; i++) {
-                if (dst[i] < 0) {
-                    dst[i] = '?';
+            int positives = StringCoding.countPositives(dst, 0, dst.length);
+            if (positives < dst.length) {
+                for (int i = positives; i < dst.length; i++) {
+                    if (dst[i] < 0) {
+                        dst[i] = '?';
+                    }
```

Comment:
I'm curious if using countPositives (and vectorization) to scan forward would be valuable for long (mostly ASCII) strings or if the method call overhead/non-constant stride is not a win for shorter strings or heavily non-ascii inputs. 

```suggestion
                for (int i = positives; i < dst.length; i = StringCoding.countPositives(dst, i + 1, dst.length - i);) {
                    if (dst[i] < 0) {
                        dst[i] = '?';
                    }
```

Reply:
There's some overhead doing countPositives calls, and doing it in a loop might provoke poor worst case performance. Im sure you can find inputs for which this is a win, though.